session/artifactmap.go

```diff
@@ -135,24 +143,29 @@ func (am *ArtifactMap) FillArtifacts(item *ItemData) error {
 
 // DeleteApp deletes an app from the ArtifactMap
 func (am *ArtifactMap) DeleteApp(appGUID string) {
-	am.DeleteItem(ResourceTypeApp, appGUID, true)
+	am.DeleteItem(ResourceTypeApp, appGUID, ArtifactEntryCompareTypeID)
```

Comment:
Just incorrect variable naming? Appid != itemid

Reply:
AppID is saved in entry.ID
ItemID is saved in entry.ItemID
so correct variable naming as far as I can see

Reply:
ArtifactEntryCompareTypeID implies item Id?

Reply:
Alright. Well, .ID is used in most other places to signify itemId so I guess that's why I thought it seemed off.

Reply:
well previous code was
```
Lookup(resourcetype, lookup string, id bool)
```
where `id` mean app GUID (in this case)

Reply:
entry.ID as artifactmap entry is always resource ID not item ID

Reply:
```
	ArtifactEntry struct {
		Name         string      `json:"name"`
		ID           string      `json:"resourceId"`
		ItemID       string      `json:"ID,omitempty"`

```

Reply:
That snippet shows the confusion very well 😆

Reply:
yea noticed the json say ID for some reason lol... well should only become json for logs so should be ok

Reply:
The way I see it, the json is the way most places refer to the itemid vs resourceid and the internal variable names are what is used in the gopherciser

Reply:
It used to be the resourceid only, the item id was a later addition.